plugins/mcreator-core/blockly/js/mcreator_extensions.js

```diff
@@ -282,6 +282,38 @@ function checkIfAfter(block, predicate) {
     return null;
 }
 
+Blockly.Extensions.registerMixin('disable_duplicate_input_type',
+    {
+        onchange: function (e) {
+            // Trigger the change only if a block is changed, moved, deleted or created
+            if (e.type !== Blockly.Events.BLOCK_CHANGE &&
+                e.type !== Blockly.Events.BLOCK_MOVE &&
+                e.type !== Blockly.Events.BLOCK_DELETE &&
+                e.type !== Blockly.Events.BLOCK_CREATE) {
+                return;
+            }
+
+            var types = new Set(); // Store the type of blocks that are already placed in a previous argument.
+            var children = this.getChildren(true); // We get all children of the block we want to check ordered, so for cases like repeating_args, the real first block is kept.
+            children.forEach(block => {
+                const type = block.type.split("_"); // We use this format: item_predicate_{typewithoutunderscores}_{optional_extra_data}
+                const realType = type[2];
+
+                if (types.has(realType)) {
+                    if (!this.isInFlyout) {
+                        const group = Blockly.Events.getGroup();

```

Comment:
It would be good to show some sort of warning for UX, similar to e.g. null_comparison_exclude_primitive_types

Reply:
I tried adding a warning, but the message never appears and I don't know why. There's already a message automatically added when the block is disabled, so I think it's ok (even though the message doesn't specify the specific reason, but the greyed out block tells the user to remove it).
<img width="517" height="90" alt="image" src="https://github.com/user-attachments/assets/213dbc5b-4b10-4b9a-a2be-943ff08b8e10" />

Reply:
Ok I found the real problem. I simply can not use parameters for the translation inside the JS file.